test/asyncworker.js

```diff
@@ -1,21 +1,85 @@
 'use strict';
 const buildType = process.config.target_defaults.default_configuration;
 const assert = require('assert');
+const async_hooks = require('async_hooks');
 
-test(require(`./build/${buildType}/binding.node`));
-test(require(`./build/${buildType}/binding_noexcept.node`));
+(async() => {
+  await test(require(`./build/${buildType}/binding.node`));
+  await test(require(`./build/${buildType}/binding_noexcept.node`));
+})();
```

Comment:
I assume this was needed to make sure the tests ran sequentially.   Could you do it without depending on await as we run the tests  with older versions of Node.js that may not have support (ex 6.x I think)